AEPAssurance/Source/AssuranceSession.swift

```diff
@@ -40,9 +40,22 @@ class AssuranceSession {
     /// indicates if Assurance SDK can start forwarding events to the session. This flag is set when a command `startForwarding` is received from the socket.
     var canStartForwarding: Bool = false
 
-    /// true indicates Assurance SDK has timeout and shutdown after non-reception of deep link URL because of which it  has cleared all the queued initial SDK events from memory.
+    /// true indicates Assurance SDK has timeout and shutdown after non-reception of deep link URL because of which it has cleared all the queued initial SDK events from memory.
     var didClearBootEvent: Bool = false
 
+    /// Boolean flag indicating whether to process and queue the SDK events listened from the wildcard listener.
+    /// This flag is set to false on the following occasions
+    ///  1. When the Assurance extension automatically shuts down on non arrival of assurance deeplink after the 5 second timeout.
+    ///  2. When the Assurance session is disconnected by the user.
+    /// This flag is turned back on when
+    ///  1. Assurance extension is reconnected to an new Assurance session
+    ///
+    /// TODO: MOB-15936
+    /// Tracking flags is difficult! This flag should be removed in favor of recreating a
+    /// new AssuranceSession for each new socket connection and making the AssuranceExtension rely
+    /// on the existence of a session for inferring event processing.
+    var canProcessSDKEvents: Bool = true
```

Comment:
rather than having to track the bool, maybe just turn it into a calculated var?  seems like the `socket.socketState` might be a good candidate.  it's `.unknown` by default, so you could just make this bool calculated by `socket.socketState != .closed && socket.socketState != .closing`

Reply:
that's a good idea, but this flag does a little more than just correlate to socket closed or closing

For example: during the shutting down of Assurance extension (after 5 secs of inactivity) we use this flag to stop processing the events. The socket state in this occasion will be `.unknown` always.

We do are happy about having more flags. But we planned to do a refactor where we can remove in favor of initiating and deleting a `AssuranceSession` instance when we need. (Today assurnaceSession instance is always initiated despite of disconnection)

Reply:
sounds good.  approving given the imminent refactor.